docs/src/examples/hamiltonian_nn.md

```diff
@@ -9,50 +9,46 @@ m\ddot x + kx = 0
 Now we make some simplifying assumptions, and assign ``m = 1`` and ``k = 1``. Analytically solving this equation, we get ``x = sin(t)``. Hence, ``q = sin(t)``, and ``p = cos(t)``. Using these solutions, we generate our dataset and fit the `NeuralHamiltonianDE` to learn the dynamics of this system.
 
 ```@example hamiltonian_cp
-using Flux, DiffEqFlux, DifferentialEquations, Statistics, Plots, ReverseDiff
+using Lux, DiffEqFlux, OrdinaryDiffEq, Statistics, Plots, Zygote, ForwardDiff, Optimisers, Random, ComponentArrays
 
-t = range(0.0f0, 1.0f0, length = 1024)
+t = range(0.0f0, 1.0f0, length=1024)
 π_32 = Float32(π)
 q_t = reshape(sin.(2π_32 * t), 1, :)
 p_t = reshape(cos.(2π_32 * t), 1, :)
 dqdt = 2π_32 .* p_t
 dpdt = -2π_32 .* q_t
 
-data = cat(q_t, p_t, dims = 1)
-target = cat(dqdt, dpdt, dims = 1)
-dataloader = Flux.Data.DataLoader((data, target); batchsize=256, shuffle=true)
-
-hnn = HamiltonianNN(
-    Flux.Chain(Flux.Dense(2, 64, relu), Flux.Dense(64, 1))
-)
-
-p = hnn.p
-
-opt = ADAM(0.01)
+data = vcat(q_t, p_t)
+target = vcat(dqdt, dpdt)
+B = 256
+dataloader = ((selectdim(data, 2, ((i-1)*B+1):(min(i * B, size(data, 2)))))
+              for i in 1:(size(data, 2)÷B))
 
-loss(x, y, p) = mean((hnn(x, p) .- y) .^ 2)
+hnn = HamiltonianNN(Lux.Chain(Lux.Dense(2, 64, relu), Lux.Dense(64, 1)))
+ps, st = Lux.setup(Random.default_rng(), hnn)
+ps_c = ps |> ComponentArray
 
-callback() = println("Loss Neural Hamiltonian DE = $(loss(data, target, p))")
+opt = ADAM(0.01f0)
```

Comment:
Can you use OptimizationOptimisers instead?

Reply:
I had run into some errors earlier with that. Let me try once again tomorrow and see.

Reply:
The latest commit uses Optimization.jl